### PR TITLE
aws_lambda: allow cross account lambda function invocation

### PR DESCRIPTION
**Pull request by vaibhav7dhage**
Signed-off-by: Vaibhav Dhage <vddhage@amazon.com>

Commit Message:
- Allow cross account lambda function invocation by setting the path in header with complete Lambda ARN instead of the function name.

Additional Description:
- Without this change, lambda filter always calls the lambda function in the AWS account where envoy runs and fails if it can't find the function in that account with the same name.

Risk Level:
- Low

Testing:
- Unit Test
- Integration Test
- Manual testing in AWS cross account setup to make sure function invocation works for both in account and cross account.

Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by suniltheta** [comment]
Can you update [changelogs](https://github.com/envoyproxy/envoy/blob/main/changelogs/current.yaml) with this as a bug fix?

File: source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc
Comment (suniltheta) [inline source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc:47]: Can we add a method inside [Arn class](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/aws_lambda/aws_lambda_filter.h#L21) so that we can directly call `arn->value()` so that we get the arn string value entirely? Then we don't have to construct the arn value every time. The config has the entire arn as a string in itself.

Also worth considering if we can simplify `class Arn`. Currently we only use `region()` & `functionName()` method inside. With this PR we can even get rid of `functionName` and other unused functions.

**Comment by vaibhav7dhage** [comment]
> Can you update [changelogs](https://github.com/envoyproxy/envoy/blob/main/changelogs/current.yaml) with this as a bug fix?

Good point!

File: source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc
Comment (vaibhav7dhage) [reply source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc:47]: Sure. Actually, planning to add another member variable `arn_` in `Arn()` class which retains the original ARN string from the config.

I would keep the other functions since they could be used in future after parsing ARN to extract specific fields.

Comment (suniltheta) [reply source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc:47]: that should work as well.

**Review by suniltheta — APPROVED** [review]
LGTM!

**Comment by wrowe** [comment]
Hi @mattklein123 , this has one review by @suniltheta and looks ready for final review, assigning as a codeowner if you could please?

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
